src/toil/test/cwl/cwlTest.py

```diff
@@ -707,6 +721,53 @@ def test_kubernetes_cwl_conformance_with_caching(self):
             ),
         )
 
+    @slow
+    @needs_wes
+    def test_wes_server_cwl_conformance(self):
+        """
+        Run the CWL conformance tests via WES. TOIL_WES_ENDPOINT must be
+        specified. If the WES server requires authentication, set TOIL_WES_USER
+        and TOIL_WES_PASSWORD.
+
+        To run manually:
+
+        TOIL_WES_ENDPOINT=http://localhost:8080 \
+        TOIL_WES_USER=test \
+        TOIL_WES_PASSWORD=password \
+        python -m pytest src/toil/test/cwl/cwlTest.py::CWLv12Test::test_wes_server_cwl_conformance -vv --log-level INFO --log-cli-level INFO
+        """
+        endpoint = os.environ.get("TOIL_WES_ENDPOINT")
+        user = os.environ.get("TOIL_WES_USER")
+        password = os.environ.get("TOIL_WES_PASSWORD")
+
+        extra_args = [f"--wes_endpoint={endpoint}"]
+        if user and password:
+            # TODO: if we run this on the CI, make sure we don't log this...
+            #  we output the entire command before we run, and cwltest logs the command if a test fails.
+            extra_args.extend([
+                f"--wes_user={user}",
+                f"--wes_password={password}"
+            ])
```

Comment:
The WES runner script should probably take these from the environment itself instead of having them set here.

Or, if we're running against local a local WES server managed by the tests, we can use no authentication and make sure we only listen on the loopback interface.